File: python/01_estrutura_sequencial/ex15.py
"""
Exercício 15

Faça um Programa que pergunte quanto você ganha por hora e o número de horas trabalhadas
no mês. Calcule e mostre o total do seu salário no referido mês, sabendo-se que são
descontados 11% para o Imposto de Renda, 8% para o INSS e 5% para o sindicato, faça um
programa que nos dê:
    a. salário bruto.
    b. quanto pagou ao INSS.
    c. quanto pagou ao sindicato.
    d. o salário líquido.
    e. calcule os descontos e o salário líquido, conforme a tabela abaixo:
        + Salário Bruto: R$
        - IR (11%): R$
        - INSS (8%) : R$
        - Sindicato (5%): R$
        = Salário Líquido: R$
        Obs: Salário Bruto - Descontos = Salário Líquido.
"""

IR = 0.11
INSS = 0.08
SINDICATO = 0.05

valor_hora = float(input("Digite o seu valor hora R$: "))
horas_trabalhadas = float(input("Digite o total de horas trabalhadas: "))
salario_bruto = valor_hora * horas_trabalhadas
valor_ir = salario_bruto * IR
valor_inss = salario_bruto * INSS
valor_sindicato = salario_bruto * SINDICATO
descontos = valor_ir + valor_inss + valor_sindicato
salario_liquido = salario_bruto - descontos
print(f"\t+ Salário Bruto: R$ {salario_bruto:.2f}")
print(f"\t- IR (11%): R$ {valor_ir:.2f}")
print(f"\t- INSS (8%): R$ {valor_inss:.2f}")
print(f"\t- Sindicato (5%): R$ {valor_sindicato:.2f}")
print(f"\t= Salário Líquido: R$ {salario_liquido:.2f}")
print(f"O total de descontos é de R$ {descontos:.2f}")

File: nodejs/01-estrutura-sequencial/ex15.js
/*
 * Exercício 15
 *
 * Faça um Programa que pergunte quanto você ganha por hora e o número de horas
 * trabalhadas no mês. Calcule e mostre o total do seu salário no referido mês,
 * sabendo-se que são descontados 11% para o Imposto de Renda, 8% para o INSS e 5% para
 * o Sindicato, faça um programa que nos dê:
 *      a. salário bruto.
 *      b. quanto pagou ao INSS.
 *      c. quanto pagou ao Sindicato.
 *      d. salário líquido.
 *      e. calcule os descontos e o salário líquido, conforme a tabela abaixo:
 *          + Salário Bruto: R$
 *          - IR (11%): R$
 *          - INSS (8%): R$
 *          - Sindicato (5%): R$
 *          = Salário Líquido: R$
 *          Obs: Salário Bruto - Descontos = Salário Líquido.
 */

import * as readlinePromises from 'node:readline/promises'
import {stdin as input, stdout as output} from 'node:process'

const rlp = readlinePromises.createInterface({input, output})

const IR = 0.11
const INSS = 0.08
const SINDICATO = 0.05

const valorHora = parseFloat(await rlp.question("Digite o seu valor hora R$: "))
const horasTrabalhadas = parseFloat(await rlp.question("Digite o total de horas trabalhadas: "))
rlp.close()

const salarioBruto = valorHora * horasTrabalhadas
const valorIR = salarioBruto * IR
const valorINSS = salarioBruto * INSS
const valorSindicato = salarioBruto * SINDICATO
const descontos = valorIR + valorINSS + valorSindicato
const salarioLiquido = salarioBruto - descontos

console.log(`\t+ Salário Bruto: R$ ${salarioBruto.toFixed(2)}`)
console.log(`\t- IR (11%): R$ ${valorIR.toFixed(2)}`)
console.log(`\t- INSS (8%): R$ ${valorINSS.toFixed(2)}`)
console.log(`\t- Sindicato (5%): R$ ${valorSindicato.toFixed(2)}`)
console.log(`\t= Salário Líquido: R$ ${salarioLiquido.toFixed(2)}`)
console.log(`O total de descontos é de R$ ${descontos.toFixed(2)}`)
